crates/router/src/utils/currency.rs

```diff
@@ -159,43 +159,43 @@ pub async fn get_forex_rates(
     state: &SessionState,
     call_delay: i64,
 ) -> CustomResult<FxExchangeRatesCacheEntry, ForexCacheError> {
-    if let Some(local_rates) = retrieve_forex_from_local().await {
+    if let Some(local_rates) = retrieve_forex_from_local_cache().await {
         if local_rates.is_expired(call_delay) {
             // expired local data
             logger::debug!("forex_log: Forex stored in cache is expired");
-            successive_fetch_and_save_forex(state, Some(local_rates)).await
+            call_forex_api_and_save_data_to_cache_and_redis(state, Some(local_rates)).await
         } else {
             // Valid data present in local
             logger::debug!("forex_log: forex found in cache");
             Ok(local_rates)
         }
     } else {
         // No data in local
-        handler_local_no_data(state, call_delay).await
+        check_local_cache_expiry_and_call_api(state, call_delay).await
     }
 }
 
-async fn handler_local_no_data(
+async fn check_local_cache_expiry_and_call_api(
```

Comment:
function says check local cache but you are checking in redis. rename it accordingly 

```suggestion
async fn call_api_if_redis_forex_data_expired(
```